### ISSUE TITLE
TypeError: m.configure is not a function

### ISSUE THREAD

**Issue by dkent600** [aurelia/framework#869]
**I'm submitting a bug report**

* **Library Version:**
1.1.5

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
v9.4.0

* **NPM Version:**
5.6.0

* **Aurelia CLI OR JSPM OR Webpack AND Version**
Running tests in jest and aurelia-testing

* **Browser:**
n/a

* **Language:**
TypeScript, JavaScript

**Current behavior:**

Please see https://github.com/aurelia/testing/issues/80 for the behavior.

Looking here:
https://github.com/aurelia/framework/blob/88a18426b799d7bd52ae6a19047ac9d4a29b451e/src/framework-configuration.js#L38

If I change `if ('configure' in m) {` to `if (m.configure) {`

the problem disappears.

`m.configure` in exists this case, but is set to `undefined`.

**Expected/desired behavior:**
no crash, more robust handling of the case, even if it is aberrant.

* **What is the motivation / use case for changing the behavior?**
To be able to run my tests.

**Comment by dkent600** [comment]
Other diagnostic info:

In this case, `moduleId` is "aurelia-templating-resources".

`m` looks like:

```
      { Compose: undefined,
        If: undefined,
        Else: undefined,
        With: undefined,
        Repeat: undefined,
        Show: undefined,
        Hide: undefined,
        HTMLSanitizer: undefined,
        SanitizeHTMLValueConverter: undefined,
        Replaceable: undefined,
        Focus: undefined,
        configure: undefined,
        AttrBindingBehavior: undefined,
        OneTimeBindingBehavior: undefined,
        OneWayBindingBehavior: undefined,
        TwoWayBindingBehavior: undefined,
        ThrottleBindingBehavior: undefined,
        DebounceBindingBehavior: undefined,
        SelfBindingBehavior: undefined,
        SignalBindingBehavior: undefined,
        BindingSignaler: undefined,
        UpdateTriggerBindingBehavior: undefined,
        AbstractRepeater: undefined,
        RepeatStrategyLocator: undefined,
        NullRepeatStrategy: undefined,
        ArrayRepeatStrategy: undefined,
        MapRepeatStrategy: undefined,
        SetRepeatStrategy: undefined,
        NumberRepeatStrategy: undefined,
        createFullOverrideContext: undefined,
        updateOverrideContext: undefined,
        getItemsSourceExpression: undefined,
        isOneTime: undefined,
        updateOneTimeBinding: undefined,
        unwrapExpression: undefined,
        viewsRequireLifecycle: undefined }
```

**Comment by Alexander-Taran** [comment]
Can you attach a zip file with your setup without node_modules ?

**Comment by dkent600** [comment]
Try this, let me know if it doesn't work.  Repo by running `npm start test`.

[repo-case.zip](https://github.com/aurelia/framework/files/1773030/repo-case.zip)

**Comment by Alexander-Taran** [comment]
works on my machine (-:

**Comment by dkent600** [comment]
@Alexander-Taran  lucky you  ;-)

Any ideas why it would work for you and not for me?  Node? NPM? NPM cache?

**Comment by dkent600** [comment]
(just kidding)

**Comment by dkent600** [comment]
The error still happens to me after I delete node_modules and package-lock.json.

**Comment by dkent600** [comment]
Did you perchance use yarn instead of npm?

**Comment by roy46** [comment]
Are you using the local install of any cli's you may have installed or a global?

I sometimes get issues like this  when I have a different global version of the aurelia-cli or webpack for instance. Bit of a long shot but if "m" is passed in from a an old version of a package it might cause this issue.

Example call to a locally installed version of cli's:
.\node_modules\.bin\au run

**Comment by dkent600** [comment]
@Alexander-Taran Great!  I'd love to hear what is the configuration error, thanks so much!

**Comment by Alexander-Taran** [comment]
Well here is where it gets weired..
you have hard lock on package versions
```
    "aurelia-loader-nodejs": "1.0.1",
    "aurelia-pal-nodejs": "1.0.0-beta.1.0.0",
    "aurelia-protractor-plugin": "1.0.3",
```

had to replace those for 
```
dev:
    "aurelia-loader-nodejs": "^1.0.1",
    "aurelia-pal-nodejs": "^1.0.0-beta.3.0.0",
    "aurelia-protractor-plugin": "^1.0.6",
```
bootstrapper also

You have not included 
` "tslib": "^1.9.0",` in dev, which is AFAIK required for decorators and metadata (tsconfig settings)


and the most weird part
i removed
```

    "moduleNameMapper": {
      "aurelia-(.*)": "<rootDir>/node_modules/aurelia-$1"
    },
```
in cli based package json it is like this 
```
   "moduleNameMapper": {
      "aurelia-(.*)": "<rootDir>/node_modules/$1"
    }, 
```

I don't know why it is there.
Works perfectly fine for me without this section at all

my jest setup from scratch:
```
  "jest": {
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "modulePaths": [
      "<rootDir>/src",
      "<rootDir>/node_modules"
    ],
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "testRegex": "\\.spec\\.(ts|js)x?$",
    "setupFiles": [
      "<rootDir>/jest-pretest.ts"
    ],
    "testEnvironment": "node"
  }
```

if you change in yours to how it is in cli - works fine..

But as jest documentation states it.. it is there for mocking out what is not needed.. So when you 'mocked' out the framework - the framework stopped being there.
That's why there were no configure.. Jest supplied an empty class instead of real module.

**Comment by dkent600** [comment]
Thanks!  It works!

I find that the cli version of this doesn't work either.  It really has to be removed completely:

```
   "moduleNameMapper": {
      "aurelia-(.*)": "<rootDir>/node_modules/$1"
    }
```

`tslib` does not seem to be required, but is good to know about.

I don't recall for sure where I got the broken configuration.  I suspect it was from [JavaScript Services](https://blogs.msdn.microsoft.com/webdev/2017/02/14/building-single-page-applications-on-asp-net-core-with-javascriptservices/), as the application originally had ASP.Net Core running server-side.  The test code originally worked, but over time as I upgraded various package versions, it broke.

@Alexander-Taran  Thanks for figuring this out!  I hope this will eventually inform various groups (Aurelia CLI and JavaScript Services) and individuals as to issues arising as time passes and dependencies evolve away from each other....
